File: cs/lc380.cs
public class RandomizedSet {

    Dictionary<int,int> map;
    List<int> ll;
    int lStart;
    int lEnd;
     Random r;
    /** Initialize your data structure here. */
    public RandomizedSet() {
        map = new Dictionary<int,int>();
        ll = new List<int>();        
        lStart = 0;
        lEnd = -1;
        r = new Random();
    }
    
    /** Inserts a value to the set. Returns true if the set did not already contain the specified element. */
    public bool Insert(int val) {
        if(!map.ContainsKey(val))
        {            
            if(lEnd == ll.Count-1)
            {//no elements removed yet.
                ll.Add(val);
                lEnd++;
            }
            else
            {
                ll[lEnd+1] = val;
                lEnd++;
            }            
            map.Add(val,lEnd);
            return true;
        }
        return false;
    }
    
    /** Removes a value from the set. Returns true if the set contained the specified element. */
    public bool Remove(int val) {
        if(map.ContainsKey(val))
        {   
            int idx = map[val];
            int temp = ll[lEnd];
            ll[lEnd] = val;
            ll[idx] = temp;
            map[temp] = idx;
            lEnd--;
            map.Remove(val);
            return true;
        }
        return false;
    }
    
    /** Get a random element from the set. */
    public int GetRandom() {
       
        
        int idx = r.Next(lStart,lEnd+1);        
        int ret = ll[idx];
      
        return ret;
    }
}

File: python/lc380.py
import random
class RandomizedSet:
   
    def __init__(self):
        self.s = {}
        self.lst = []


    def insert(self, val: int) -> bool:
        if val in self.s:
            return False
        else:
            self.lst.append(val)
            self.s[val] = len(self.lst) - 1            
            return True

    def remove(self, val: int) -> bool:
        # print('remove ', val, self.s)
        if val in self.s:
            rem_idx = self.s[val]
            del self.s[val]
            if rem_idx != len(self.lst) - 1:
                update_val = self.lst[len(self.lst) - 1]
                self.lst[rem_idx] = self.lst[len(self.lst) - 1]
                self.s[update_val] = rem_idx
            self.lst.pop()
            return True
        else:
            return False

    def getRandom(self) -> int:        
        rnd_idx = random.randint(0, len(self.lst) - 1)
        return self.lst[rnd_idx]
